install/operator/pkg/apis/syndesis/v1alpha1/syndesis_types.go

```diff
@@ -119,6 +119,8 @@ type DbConfiguration struct {
 	User                 string              `json:"user,omitempty"`
 	Database             string              `json:"database,omitempty"`
 	ImageStreamNamespace string              `json:"imageStreamNamespace,omitempty"`
+	// Options for configuring an external database instead of deploying one
+	ExternalDBSecret string `json:"externalDBSecretRef,omitempty"`
```

Comment:
You should not put secrets in a CR since they are not encoded.

Reply:
I'm not sure I understand? This is a reference to a secret name, not the secret itself

Reply:
ah.. if it's a ref, then that's ok.  Note that your json name has to match  the go field name or else you'll get an error when you generate the openapi docs.

But then again, we already have a secret where we store the DB password.  Why don't we keep using it?

Reply:
I think that secret is `syndesis-global-config`, right?

The reason I went with not reusing that is that it contains other information, almost all the content of that secret makes sense to be created by the operator in all cases. So, even though there ends up being some duplication, this way `syndesis-global-config` stays "owned" and maintained by the operator, and the user-created secret only has connection information and stays "owned" by the user.

However, this made more sense when keeping all connection info about the Postgres instance was going into a secret. So I'm happy to change it if needed.